src/pykeen/pipeline.py

```diff
@@ -834,9 +834,11 @@ def pipeline(  # noqa: C901
     if not metadata:
         metadata = {}
     title = metadata.get('title')
+    tags = metadata.get('run_tags', None)
```

Comment:
It would make more sense that this result tracker-specific information is passed to the pipeline via the `result_tracker_kwargs` instead of the `metadata`.

https://github.com/pykeen/pykeen/blob/2c9101c41d1f757a30d5b57cb0d7650f7f60f671/src/pykeen/pipeline.py#L832

Tags are only relevant for MLflow, so it wouldn't necessitate adding new functionality to the base class

Reply:
I would suggest instead modifying the `__init__()` function for the `MLFlowResultTracker` to accept a `tags` keyword argument and saving it as an instance variable.

https://github.com/pykeen/pykeen/blob/2c9101c41d1f757a30d5b57cb0d7650f7f60f671/src/pykeen/trackers.py#L48-L74

Reply:
okay yeah - makes sense.. and then in the start_run - after starting the run - update tags as well? I only basically added the base class method, because you can't add tags until there is an active run.

Reply:
maybe there's just a bit of confusion since the `title` comes from the metadata - this is a pretty general piece of information (not specific to result tracker). Maybe it would make more sense if there's a priority to get the title from the `result_tracker_kwargs` first then the `metadata` second (I could take care of this in another PR - it's not relevant to yours)

Reply:
> @sunny1401 you'll see that I made a few updates:
> 
> 1. The double underscore in the variable name `self.__tags` isn't necessary. Usually this only makes code harder to read, so I've changed it to just `tags`
> 2. Rather than using two different lines to initialize `self.tags` that includes a conditional, now it is just set to whatever is given in the call to `__init__`.
> 3. A none check is now used instead of a length check.
> 
> As a side note, in general, dictionaries can also be checked for truthyness or falsiness, so you don't have to check the length being not equal to or equal to zero. You will see that:
> 
> ```python
> def check_truthy(d):
>     if d:
>        print('truthy dict')
>     else:
>         print('falsy dict')
> 
> check_truthy({})
> 'falsy dict'
> 
> check_truthy({'hello': 5})
> 'truthy dict'
> ```

I saw the changes.. Makes total sense. I am sorry I got confused about this.